python/ray/_private/function_manager.py

```diff
@@ -389,14 +389,22 @@ def _wait_for_function(self, function_descriptor, job_id, timeout=10):
         warning_sent = False
         while True:
             with self.lock:
-                if self._worker.actor_id.is_nil() and (
-                    function_descriptor.function_id in self._function_execution_info
-                ):
-                    break
-                elif not self._worker.actor_id.is_nil() and (
-                    self._worker.actor_id in self._worker.actors
-                ):
+                if self._worker.actor_id.is_nil():
+                    if function_descriptor.function_id in self._function_execution_info:
+                        break
+                    else:
+                        key = make_function_table_key(
+                            b"RemoteFunction",
+                            job_id,
+                            function_descriptor.function_id.binary(),
+                        )
+                        if self.fetch_and_register_remote_function(key) is True:
```

Comment:
Does this busy loop if fetching fails?

Reply:
It's ok since the worker will not make progress anyway before